app/models/spree/payment.rb

```diff
@@ -181,15 +181,25 @@ def revoke_adjustment_eligibility
       )
     end
 
+    # rubocop:disable Metrics/AbcSize, Metrics/LineLength
     def validate_source
       if source && !skip_source_validation && !source.valid?
-        source.errors.each do |field, error|
-          field_name = I18n.t("activerecord.attributes.#{source.class.to_s.underscore}.#{field}")
-          errors.add(Spree.t(source.class.to_s.demodulize.underscore), "#{field_name} #{error}")
+        if Gem::Requirement.new(">= 6.1").satisfied_by?(Rails.gem_version)
```

Comment:
Why do we need this? Our current Rails version is 6.1.4 and we are not planning on going backwards.